frontend/src/metabase/dashboard/components/DashCard/DashCard.styled.tsx

```diff
@@ -6,6 +6,7 @@ export interface DashCardRootProps {
   isNightMode: boolean;
   isUsuallySlow: boolean;
   hasHiddenBackground: boolean;
+  shouldAutoPreviewInEditing: boolean;
```

Comment:
This is a generic DashCard which can be a line chart that does not have preview/editing states so it is not a generic attribute. I'd suggest moving all text cards' specific logic inside their components

Reply:
This is placed inside the generic DashCard b/c `box-shadow` and `background` are DashCard applied styles. If we want Text/Heading components to be able to auto-preview, we need to be able to change the DashCard styling that gets applied to them while they are editing. Otherwise, if we were to auto-preview 
1. a `<Heading />` component or 
2. a `<Text />` component with background disabled

it would still show a white background and box-shadow around the card since those are generically applied to DashCards.

**With DashCard specific styling**
![image](https://github.com/metabase/metabase/assets/22608765/93038cde-b68f-46a3-9e11-648d9ab34928)

**Without DashCard styling**
![image](https://github.com/metabase/metabase/assets/22608765/7a0f9399-ee6e-4e1f-a520-01eeeb468fb8)

As you can see heading components and text components w/ background disabled should never show a background or have a box-shadow, but they do if the styling is not applied to the DashCard component.

If you have a idea that solves this issue, please let me know :heart: